tests/integration/pkg/helpers/api_rule.go

```diff
@@ -90,17 +91,17 @@ func ApplyApiRuleV2Alpha1(toExecute RetryableApiRule, onRetry RetryableApiRule,
 			if err != nil {
 				return err
 			}
-			if apiStatus.Status.State != "Ready" {
-				log.Println("APIRule status not Ready: " + apiStatus.Status.Description)
-				return errors.New("APIRule status not Ready: " + apiStatus.Status.Description)
+			if apiStatus.Status.State == "Error" {
+				log.Printf("APIRule status is Error: %s", apiStatus.Status.Description)
+				return fmt.Errorf("APIRule status is Error: %s", apiStatus.Status.Description)
```

Comment:
Error should not be capitalized https://go.dev/wiki/CodeReviewComments#error-strings